chroma-common/blockdevices/blockdevice_zfs.py

```diff
@@ -95,14 +105,50 @@ def __exit__(self, type, value, traceback):
             self.unlock_pool()
 
     def lock_pool(self):
-        lock = self.import_locks[self.pool_path]
-        lock.acquire()
-        # log.debug('ZfsDevice acquired lock on %s' % self.pool_path)
+        """
+        Attempt to acquire a lock on a given zpool through a lockfile, increment the reference count each time
+        this method is called, regardless of whether acquire is actually called. Keep trying until we know
+        we are locking the zpool.
+
+        Lock acquire polls the timeout at intervals of LOCK_ACQUIRE_TIMEOUT.
+        """
+        if ZfsDevice.locks_dir_initialized == False:
+            # ensure lock directory exists
+            try:
+                os.mkdir(self.ZPOOL_LOCK_DIR)
+            except OSError:
+                pass
+
+            ZfsDevice.locks_dir_initialized = True
+
+        while not self.lock.i_am_locking():
```

Comment:
Should there not be some kind of timeout here to deal with the unexpected situation that trying to acquire the lock takes longer than the caller expected, or never completes at all, particularly given that we are holding locks around things (i.e. zfs operations) that we don't control?

From a troubleshooting perspective, it would be much more useful for example for a log message or notification to show up saying that locking failed due to a timeout rather than having a locking call waiting endlessly.